modules/auxiliary/analyze/crack_aix.rb

```diff
@@ -0,0 +1,242 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/auxiliary/password_cracker'
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Auxiliary::PasswordCracker
+
+  def initialize
+    super(
+      'Name'            => 'Password Cracker: AIX',
+      'Description'     => %Q{
+          This module uses John the Ripper or Hashcat to identify weak passwords that have been
+        acquired from passwd files on AIX systems.  These utilize DES hashing.
+        DES is format 1500 in Hashcat.
+        DES is descrypt in JTR.
+      },
+      'Author'          =>
+        [
+          'theLightCosine',
+          'hdm',
+          'h00die' # hashcat integration
+        ] ,
+      'License'         => MSF_LICENSE,  # JtR itself is GPLv2, but this wrapper is MSF (BSD)
+      'Actions'         =>
+        [
+          ['john', {'Description' => 'Use John the Ripper'}],
+          ['hashcat', {'Description' => 'Use Hashcat'}],
+        ],
+      'DefaultAction' => 'john',
+    )
+
+  end
+
+  def show_command(cracker_instance)
+    if datastore['ShowCommand']
+      if action.name == 'john'
+        cmd = cracker_instance.john_crack_command
+      elsif action.name == 'hashcat'
+        cmd = cracker_instance.hashcat_crack_command
+      end
+      print_status("   Cracking Command: #{cmd.join(' ')}")
+    end
+  end
+
+  def print_results(tbl, cracked_hashes)
+    cracked_hashes.each do |row|
+      unless tbl.rows.include? row
+        tbl << row
+      end
+    end
+    tbl.to_s
+  end
+
+  def run
+    def process_crack(results, hashes, core_id, username, password, hash_type, method)
```

Comment:
note to self, combine the user data into a hash...
`results, hashes, userhash, hash_type, method`